util/src/main/scala/org/ensime/util/ensimefile.scala

```diff
@@ -26,7 +26,7 @@ package ensimefile {
 
     /** Direct access contents: not efficient for streaming. */
     def readStringDirect()(implicit cs: Charset): String
-
+    def toPath: Path
```

Comment:
I'm not keen on this change. It is very difficult to deal with `File` vs "file in an archive" and the point of this sealed family was to force us to deal with both cases separately when it comes up. You may have to rethink all the callers of `toPath` to see what you really want out of the API. Do you perhaps want a string identifier? In which case `uriString` is perhaps what you want.